lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -342,6 +331,12 @@ bool ProxyRenderDelegate::_Populate() {
         MProfilingScope subProfilingScope(HdVP2RenderDelegate::sProfilerCategory,
             MProfiler::kColorD_L1, "Populate");
 
+        // if previously populated, destroy the old delegate to be able to populate the new excluded paths
```

Comment:
Given we already use ProxyRenderDelegate::_InitRenderDelegate as the only place to detect changes and re-initialize VP2RenderDelegate. How about we leave the creation of scene delegate in there (so no need for _ResetSceneDelegate method) but similar to how IsUsdStageUpToDate check is done in _InitRenderDelegate, we add a check for IsExcludePrimsUpToDate to reset the scene delegate? 

@williamkrick Does this match your thinking?

Reply:
I agree we should keep all of those checks & re-initialization together and they are already in _InitRenderDelegate(). If we can simplify the expression in _Populate to just check _isPopulated and then do the work then I would say move the destructive part out to _initRenderDelegate along with the "extra" part of the expression testing IsUsdStageUpToDate() and IsExcludePrimsUpToDate(). If we can't clean up the expression we might have to re-think the method we have and how we are organizing the code.

The documentation for _InitRenderDelegate will need to be updated, right now it says one time initialization but it is already not a one-time thing. Maybe we should also rename that function to reflect that fact that it is not just initialization but also responding to changes that require re-initialization?

I suppose the real question we are answering is how do we want to organize the code. Should we have a function per object or a function per task? I think what is there is more like a function per task approach, so moving the re-initialize stuff out makes sense.